### PR TITLE
deal with repeated patterns inside a word

### PR DESCRIPTION
**Pull request by jaumeortola**

### CODE REVIEW

File: hyphen.js [hunk 1]
```diff
@@ -35,25 +35,32 @@
...
+        if (patternFits) {
+          p.push(patternData.pattern + ">" + patternData.levels.join(""));
+
+          for (var i = 0; i < patternData.levels.length; i++)
+            levels[patternEntityIndex + i] = Math.max(
+              patternData.levels[i],
+              levels[patternEntityIndex + i]
+            );
+        }
+        if (patternEntityIndex > -1 && patternData.text.length > 0) {
```
Comment (krisztianb) [inline hyphen.js:59]: I think patternData.text should never be 0, or am I mistaken?

Comment (jaumeortola) [reply hyphen.js:59]: There are empty patterns in all languages. I don't know why. See: https://github.com/ytiurin/hyphen/blob/master/patterns/de.js#L24624
If these empty patterns are removed or ignored at some previous point, then this condition can be removed.

Comment (krisztianb) [reply hyphen.js:59]: If they are not needed, they should be removed and the matching code needn't consider them.
Then we could move the else part of your last if/else logic above "patternFits", which doesn't have to be calculated if the patternEntityIndex is -1.
In the end we will have to wait for @ytiurin to comment and merge the changes.

**Comment by ytiurin** [comment]
Sorry for waiting so long @jaumeortola @krisztianb

**Comment by krisztianb** [comment]
@ytiurin Can you tell us why there are empty patterns in the language files? We had a little discussion about this with @jaumeortola as you can see above?

**Comment by ytiurin** [comment]
@krisztianb js patterns are traspiled from tex patterns with tex2js script in /tex folder, it's probably a bug there, I can take a look

**Comment by krisztianb** [comment]
@ytiurin I was just asking, because the code could be simplified if those are not present.

**Pull request merged**